src/audio_unit/delay.rs

```diff
@@ -1,32 +1,102 @@
 use crate::{ring_buffer, AudioUnit, Result};
+use anyhow::anyhow;
 use cpal::StreamConfig;
 use ringbuf::{Consumer, Producer, RingBuffer};
+use wmidi::MidiMessage;
+
+pub type DelayMs = u32;
+
+type MidiMessageHandler = Box<dyn Fn(&[MidiMessage<'static>]) -> Option<DelayMs> + Send + Sync>;
```

Comment:
Passing around closures makes for some gnarly types. Like things in arrays, the closure field here needs to be boxed because the size is not known.